Implement auto-merge bot functionality

mbestavros opened the pull request:
This code implements auto-merge for the Enarx organization bot.

While functionally complete, I'm submitting in draft form until I know the name of the encrypted secret to use (currently `secrets.BOT_TOKEN`).

This code should run every 15 minutes, and will leave logs about which PRs were and were not auto-merged.

npmccallum commented on .github/workflows/scheduled-run.yml:
```diff
@@ -0,0 +1,16 @@
+jobs:
+  scheduled-run:
+    runs-on: ubuntu-latest
+    container: quay.io/enarx/fedora
```
We probably don't want to use our container for this. It is rather large.

mbestavros replied:
I suppose the only things we really need are Python and PyGithub. Makes sense to have a container with only those things included.

npmccallum replied:
If `ubuntu-latest` has `PyGithub`, we probably don't need a container at all.

mbestavros replied:
Very good point. At worst, we'd probably have to `pip install` it, which would still take less time than spinning up a container would.

npmccallum commented on .github/workflows/scheduled-run.yml:
```diff
@@ -0,0 +1,16 @@
+    - name: scheduled-run
+      env:
+        BOT_TOKEN: ${{ secrets.BOT_TOKEN }}
+      run: scripts/exec $BOT_TOKEN "enarx" $GITHUB_EVENT_PATH
```
```suggestion
      run: scripts/exec $BOT_TOKEN "enarx"
```

npmccallum commented on scripts/exec:
```diff
@@ -0,0 +1,41 @@
+from github import Github
+
+# Get inputs from shell
+(token, organization_name, path) = sys.argv[1:4]
```
```suggestion
(token, organization_name) = sys.argv[1:4]
```

npmccallum commented on scripts/exec:
```diff
@@ -0,0 +1,41 @@
+        pr_reviews = pr.get_reviews()
+        
+        # Parse reviews for merge criteria.
+        changes_requested = False
```
```suggestion
        changes_requested = 0
```

npmccallum commented on scripts/exec:
```diff
@@ -0,0 +1,41 @@
+            if review.state == "APPROVED":
+                approving_reviews += 1
+            if review.state == "CHANGES_REQUESTED":
+                changes_requested = True
```
```suggestion
                changes_requested += 1
```

npmccallum commented on scripts/exec:
```diff
@@ -0,0 +1,41 @@
+                changes_requested = True
+        
+        # If all merge criteria pass, merge the PR.
+        if pr.mergeable == True and approving_reviews >= 2 and changes_requested == False:
```
```suggestion
        if pr.mergeable == True and approving_reviews >= 2 and changes_requested == 0:
```

npmccallum commented on scripts/exec:
```diff
@@ -0,0 +1,41 @@
+        changes_requested = False
+        approving_reviews = 0
+        for review in pr_reviews:
+            if review.state == "APPROVED":
```
What other states are there? There could be subtle bugs here...

mbestavros replied:
There is surprisingly no documentation on the `state` field for `PullRequestReview` on Github's end: https://developer.github.com/v3/pulls/reviews/

They do provide examples in the sample responses, but no definitive listing of all possible states. The ones I can see are:

- APPROVED
- CHANGES_REQUESTED
- PENDING
- DISMISSED

The last one might cause some problems with our criteria that all "Changes Requested" reviews be updated to "Approved", though I don't know exactly how that state works.

npmccallum commented on scripts/exec:
```diff
@@ -0,0 +1,41 @@
+        
+        # If all merge criteria pass, merge the PR.
+        if pr.mergeable == True and approving_reviews >= 2 and changes_requested == False:
+            print("Merging PR " + str(pr.number) + " from " + repo.name)
```
For this, and all other strings, use a [format specifier](https://pyformat.info) or [f-string](https://realpython.com/python-f-strings/) (preferred).

mbestavros commented:
Closing in favor of https://github.com/enarxbot/enarxbot/pull/2. I committed the cardinal sin of making a PR from master...

@npmccallum I've addressed all the minor feedback from your first review, and moved the larger points over to the new PR.
